### ISSUE TITLE
default styles for td, tr, th

### ISSUE THREAD

**Issue by tobibeer** [Jermolene/TiddlyWiki5#2018]
In [tb5](http://tobibeer.github.io/tb5), I define the default vertical alignment for table cells as...

```
tr, td, th {
vertical-align:top;
}
```

Is there / what is the appropriate place to define default styles for all themes?

An example for the problem with the default:

![center](https://cloud.githubusercontent.com/assets/97566/10455545/48d64f72-71bc-11e5-883e-148cd36d6035.jpg)

With the proposed change...

![top](https://cloud.githubusercontent.com/assets/97566/10455546/48edfe06-71bc-11e5-99b0-60296d9f0cba.jpg)

**Comment by Jermolene** [comment]
Hi @tobibeer I think this would be rather a backwards incompatible change.

Instead, perhaps we should introduce a few useful, core CSS classes for standard core table styles to cover common use cases?

**Comment by tobibeer** [comment]
> I think this would be rather a backwards incompatible change.

Possibly, although I'm not sure anyone relies on vertical align in tables (or should for layouting purposes)... but yes, possibly.

> Instead, perhaps we should introduce a few useful, core CSS classes for standard core table styles to cover common use cases?

Sure. How about `table.tc-top`? Perhaps another also `table.tc-borderless`?

**Comment by Jermolene** [comment]
> Sure. How about table.tc-top? Perhaps another also table.tc-borderless?

I'm not sure it's worth a class just to encapsulation the top alignment. I was thinking of higher level classes, kind of like the choices one gets in PowerPoint. But combinations of classes could also work I guess.

**Comment by tobibeer** [comment]
> I was thinking of higher level classes, kind of like the choices one gets in PowerPoint. 

Yes, not necessarily just bound to table cells, of course. Just wanted to indicate that we'd have to be specific to cover certain table elements.

How about a subset of (Eric's StyleSheetShortcuts)[http://tiddlytools.com/#StyleSheetShortcuts]?

**Comment by Jermolene** [comment]
> How about a subset of Eric's StyleSheetShortcuts?

That's certainly one approach, and has the advantage of being tried and trusted. I had a look for open source libraries of different table themes, but could find anything. Most of the CSS frameworks that I looked at just support a single table style.

**Comment by tobibeer** [comment]
To be clear, we're just aiming at often used important defaults here. Right?

**Comment by pmario** [comment]
IMO TW should have an absolute minimum set of CSS default rules.

Something like StyleSheetShortcuts should be a plugin, theme or palette. Wherever it fits best.

**Comment by tobibeer** [comment]
> Something like StyleSheetShortcuts should be a plugin, theme or palette. Wherever it fits best.

Surely neither a theme nor a palette. It could be wrapped up as a plugin, but then I'm not sure it's worth it. It's appears to be rather a simple StyleSheet that is available independent / on top of any plugins, themes or palettes.

However, I'm not sure that it is to make amends for and overwrite otherwise problematic defaults like in the OP. I think "fixing" or changing those is independent from a simple extension with meaningful base-classes.

**Comment by pmario** [comment]
Yes. Table defaults should be part of "base" or "vanilla" CSS

**Comment by pmario** [comment]
There are some [table utility classes](https://tiddlywiki.com/#Tables%20in%20WikiText%20CSS%20Utility%20Classes) that are used by the core.

They are documented, so users know how to define their own styles if they need to.
